pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -543,6 +543,15 @@ def _get_tools(self, model_request_parameters: ModelRequestParameters) -> list[c
     def _get_web_search_options(self, model_request_parameters: ModelRequestParameters) -> WebSearchOptions | None:
         for tool in model_request_parameters.builtin_tools:
             if isinstance(tool, WebSearchTool):  # pragma: no branch
+                # Check if the current model supports web search using the model profile
+                if not OpenAIModelProfile.from_profile(self.profile).openai_supports_web_search:
+                    raise WebSearchNotSupportedError(
+                        f"WebSearchTool is not supported with model '{self.model_name}'. "
+                        f'Please use one of the following options:\n'
+                        f'1. Use a supported ChatCompletion model: gpt-4o-search-preview, gpt-4o-mini-search-preview\n'
+                        f'2. Use OpenAI Responses API instead: OpenAIResponsesModel()'
```

Comment:
No need to suggest using an older model, which is almost never desirable.

```suggestion
                        'Please use `OpenAIResponsesModel` instead.'
```